ledger/src/blocktree_processor.rs

```diff
@@ -535,7 +610,8 @@ fn process_pending_slots(
         let allocated = thread_mem_usage::Allocatedp::default();
         let initial_allocation = allocated.get();
 
-        if process_single_slot(blocktree, &bank, &last_entry_hash, opts).is_err() {
+        let slot_result = process_single_slot(blocktree, &bank, &last_entry_hash, opts, recyclers);
+        if is_result_fatal(&slot_result) {
```

Comment:
Can you double check this change? Before we would `continue` for any error result. But now, in the case where an error is not fatal, we don't and instead root the slot.